<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
From f034b1b0f33a7bd4dde23f0bc1fa8e00e3518c9d Mon Sep 17 00:00:00 2001
=======
From 2a4b87bb00805156d40782a3197fbbf20937da6f Mon Sep 17 00:00:00 2001
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:15:31 -0800
Subject: [PATCH] Add an option to disable lib2to3

lib2to3 is a library to convert Python 2.x code to Python 3.x. As
such, it is probably not very useful on embedded system targets.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
[ Bernd Kuhls: ported to Python 3.11.4]
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
[ Adam Duskett: ported to Python 3.12.1 ]
Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
 Makefile.pre.in | 17 ++++++++++++-----
 configure.ac    |  5 +++++
 setup.py        |  6 +++---
 3 files changed, 20 insertions(+), 8 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 28cf88e..63fa9fb 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1395,7 +1395,9 @@ ifeq (@PYDOC@,yes)
=======
 Makefile.pre.in | 19 ++++++++++++++-----
 configure.ac    |  6 ++++++
 2 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 401d130b804..b44ad736144 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2054,7 +2054,9 @@ ifeq (@PYDOC@,yes)
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s pydoc$(VERSION) pydoc3)
 endif
 	-rm -f $(DESTDIR)$(BINDIR)/2to3
+ifeq (@LIB2TO3@,yes)
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s 2to3-$(VERSION) 2to3)
+endif
 	if test "x$(LIPO_32BIT_FLAGS)" != "x" ; then \
 		rm -f $(DESTDIR)$(BINDIR)/python3-32$(EXE); \
 		(cd $(DESTDIR)$(BINDIR); $(LN) -s python$(VERSION)-32$(EXE) python3-32$(EXE)) \
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
@@ -1441,7 +1443,6 @@ LIBSUBDIRS=	asyncio \
=======
@@ -2101,7 +2103,6 @@ LIBSUBDIRS=	asyncio \
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
 		idlelib idlelib/Icons \
 		importlib importlib/metadata \
 		json \
-		lib2to3 lib2to3/fixes lib2to3/pgen2 \
 		logging \
 		multiprocessing multiprocessing/dummy \
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
 		site-packages \
@@ -1458,10 +1459,6 @@ LIBSUBDIRS=	asyncio \
 TESTSUBDIRS=	ctypes/test \
 		distutils/tests \
 		idlelib/idle_test \
-		lib2to3/tests \
-		lib2to3/tests/data \
-		lib2to3/tests/data/fixers \
-		lib2to3/tests/data/fixers/myfixes \
 		sqlite3/test \
 		test test/audiodata \
 		test/capath test/cjkencodings \
@@ -1535,6 +1532,14 @@ ifeq (@PYDOC@,yes)
=======
 		re \
@@ -2208,10 +2209,6 @@ TESTSUBDIRS=	idlelib/idle_test \
 		test/test_importlib/resources/zipdata02 \
 		test/test_importlib/source \
 		test/test_json \
-		test/test_lib2to3 \
-		test/test_lib2to3/data \
-		test/test_lib2to3/data/fixers \
-		test/test_lib2to3/data/fixers/myfixes \
 		test/test_module \
 		test/test_peg_generator \
 		test/test_pydoc \
@@ -2268,6 +2265,14 @@ ifeq (@PYDOC@,yes)
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
 LIBSUBDIRS += pydoc_data
 endif
 
+ifeq (@LIB2TO3@,yes)
+LIBSUBDIRS += lib2to3 lib2to3/fixes lib2to3/pgen2
+TESTSUBDIRS += test/test_lib2to3			\
+		test/test_lib2to3/data \
+		test/test_lib2to3/data/fixers \
+		test/test_lib2to3/data/fixers/myfixes
+endif
+
 TEST_MODULES=@TEST_MODULES@
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
 libinstall:	build_all $(srcdir)/Modules/xxmodule.c
 	@for i in $(SCRIPTDIR) $(LIBDEST); \
@@ -1637,10 +1642,12 @@ ifeq (@PYC_BUILD@,yes)
 		-j0 -d $(LIBDEST)/site-packages -f \
=======
 
 .PHONY: libinstall
@@ -2351,10 +2356,12 @@ ifeq (@PYC_BUILD@,yes)
 		-o 0 -o 1 -o 2 $(COMPILEALL_OPTS) -d $(LIBDEST)/site-packages -f \
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
 		-x badsyntax $(DESTDIR)$(LIBDEST)/site-packages
 endif
+ifeq (@LIB2TO3@,yes)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -m lib2to3.pgen2.driver $(DESTDIR)$(LIBDEST)/lib2to3/Grammar.txt
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -m lib2to3.pgen2.driver $(DESTDIR)$(LIBDEST)/lib2to3/PatternGrammar.txt
+endif
 
 # bpo-21536: Misc/python-config.sh is generated in the build directory
 # from $(srcdir)Misc/python-config.sh.in.
@@ -2480,7 +2487,9 @@ libainstall: all scripts
 	$(INSTALL_SCRIPT) $(srcdir)/install-sh $(DESTDIR)$(LIBPL)/install-sh
 	$(INSTALL_SCRIPT) python-config.py $(DESTDIR)$(LIBPL)/python-config.py
 	$(INSTALL_SCRIPT) python-config $(DESTDIR)$(BINDIR)/python$(LDVERSION)-config
+ifeq (@LIB2TO3@,yes)
 	$(INSTALL_SCRIPT) $(SCRIPT_2TO3) $(DESTDIR)$(BINDIR)/2to3-$(VERSION)
+endif
 	$(INSTALL_SCRIPT) $(SCRIPT_IDLE) $(DESTDIR)$(BINDIR)/idle$(VERSION)
 ifeq (@PYDOC@,yes)
 	$(INSTALL_SCRIPT) $(SCRIPT_PYDOC) $(DESTDIR)$(BINDIR)/pydoc$(VERSION)
diff --git a/configure.ac b/configure.ac
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
index 9079531..34c2ba9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6014,6 +6014,11 @@ else
 fi
 AC_SUBST(TEST_MODULES)
=======
index ca226dcc5b9..9b77582560d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7572,6 +7572,12 @@ PY_STDLIB_MOD([xxlimited_35], [test "$with_trace_refs" = "no"], [test "$ac_cv_fu
 # substitute multiline block, must come after last PY_STDLIB_MOD()
 AC_SUBST([MODULE_BLOCK])
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch
 
+AC_SUBST(LIB2TO3)
+
+AC_ARG_ENABLE(lib2to3,
+	AS_HELP_STRING([--disable-lib2to3], [disable lib2to3]),
+	[ LIB2TO3="${enableval}" ], [ LIB2TO3=yes ])
 
 # generate output files
<<<<<<< HEAD:package/python3/0012-Add-an-option-to-disable-lib2to3.patch
 AC_CONFIG_FILES(Makefile.pre Misc/python.pc Misc/python-embed.pc Misc/python-config.sh)
diff --git a/setup.py b/setup.py
index d23f148..663fd44 100644
--- a/setup.py
+++ b/setup.py
@@ -2722,11 +2722,11 @@ def main():
     import warnings
     warnings.filterwarnings("ignore",category=DeprecationWarning)
 
-    scripts = ['Tools/scripts/idle3', 'Tools/scripts/2to3',
-               'Lib/smtpd.py']
+    scripts = ['Tools/scripts/idle3', 'Lib/smtpd.py']
     if not '--disable-pydoc' in sysconfig.get_config_var("CONFIG_ARGS"):
         scripts += [ 'Tools/scripts/pydoc3' ]
-
+    if not '--disable-lib2to3' in sysconfig.get_config_var("CONFIG_ARGS"):
+        scripts += [ 'Tools/scripts/2to3' ]
     setup(# PyPI Metadata (PEP 301)
           name = "Python",
           version = sys.version.split()[0],
-- 
2.30.2
=======
 AC_CONFIG_FILES(m4_normalize([
   Makefile.pre
-- 
2.39.5
>>>>>>> upstream-2025.02.7:package/python3/0004-Add-an-option-to-disable-lib2to3.patch

